cve_bin_tool/output_engine/console.py

```diff
@@ -52,19 +80,37 @@ def output_console(
 
     cve_by_remarks: DefaultDict[Remarks, List[Dict[str, str]]] = defaultdict(list)
     # group cve_data by its remarks
-    for product_info, cve_data in all_cve_data.items():
-        for cve in cve_data["cves"]:
-            cve_by_remarks[cve.remarks].append(
-                {
-                    "vendor": product_info.vendor,
-                    "product": product_info.product,
-                    "version": product_info.version,
-                    "cve_number": cve.cve_number,
-                    "severity": cve.severity,
-                    "score": cve.score,
-                    "cvss_version": cve.cvss_version,
-                }
-            )
+    if affected_versions != 0:
```

Comment:
This `if...else` is just a huge chunk of duplication, I'm not sure how much it would affect performance but I would put this `if` in the inner loop and `affected_versions` to the last CVE if needed. But only if it won't slow down things much of course.